### PR TITLE
Demo notebook with a narrative

### PR DESCRIPTION
**Pull request by briangodsey**
This PR adds a notebook in the `examples` folder. This new notebook is intended to show a simple narrative for potential users who may be wondering why and how to start using Kaskada in a meaningful way.

This may be a developer's first look at Kaskada, and thus should be self-contained and provide a simple but meaningful user story that conveys some idea of why and when to use Kaskada.

Questions and comments are more than welcome!

### CODE REVIEW

File: examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+    "event_count_total: DemoEvents | count(),\n",
+    "purchases_total_count: DemoEvents | when(DemoEvents.event_name == 'purchase') | count(),\n",
+    "revenue_total: DemoEvents.revenue | sum(),\n",
+    "purchases_daily: DemoEvents | when(DemoEvents.event_name == 'purchase') | count(window=since(daily())),\n",
+    "revenue_daily: DemoEvents.revenue | sum(window=since(daily())),\n",
+    "event_count_hourly: DemoEvents | count(window=since(hourly())),\n",
+    "\n",
+    "```\n",
+    "\n",
+    "Of course, a few more lines of code is needed to put these calculations to work,\n",
```
Comment (therapon) [inline examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb:46]:
```suggestion
    "Of course, a few more lines of code are needed to put these calculations to work,\n",
```

File: examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bbe86c2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install --upgrade pip\n",
+    "%pip install kaskada==0.1.1a7\n",
```
Comment (therapon) [inline examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb:87]: I think we just released a non-alpha version so we might want to use 

```
pip install kaskada
```

File: examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f966d56a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# the object `event_calculations` has an attribute called `dataframe` can be\n",
```
Comment (therapon) [inline examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb:442]:
```suggestion
    "# the object `event_calculations` has an attribute called `dataframe` that can be\n",
```

File: examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb [hunk 4]
```diff
@@ -0,0 +1,559 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d818e1f1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: remove when repo goes public. While the repo is still private, we need a\n",
```
[the remark this answers is not in the document]
Comment (briangodsey) [reply examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb:111]: yep... I was going to confirm. will do!

File: examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb (hunk 2 above)
Comment (briangodsey) [reply examples/Kaskada Demo for Event Processing and Time-centric Calculations.ipynb:87]: makes sense!

**Pull request merged**